feat: update braze dependency to v6 (#62)
mparticle-integrations/mparticle-javascript-integration-braze: README.md

![Braze Logo](https://github.com/mparticle-integrations/mparticle-javascript-integration-appboy/blob/master/braze-logo.png) 

⚠️⚠️⚠️
# Notice! Opt in is now available for Braze Web SDK V6 - Action Required

You can now select what version of the Braze SDK you want to use when setting up a Braze connection in the mParticle UI.  Braze occasionally makes breaking changes to their SDK, so if you call `braze` directly in your code, you will have to update your code to ensure your website performs as expected when updating versions of Braze.

> Please review the [Braze Changelog](https://www.braze.com/docs/developer_guide/platform_integration_guides/web/changelog#600) and [V6 migration guide](https://github.com/braze-inc/braze-web-sdk/blob/master/UPGRADE_GUIDE.md) to learn about the differences between V5 and V6 and what changes you will need to make in your code. There may be API changes that affect you if you call `braze` directly from your code.

Reviewer: The most significant breaking change is the removal of the legacy News Feed feature and associated methods, in addition to the removal and renaming of several APIs.

You can opt into the latest major version of the Braze Web SDK whether you implement mParticle's Web SDK using npm or our snippet/CDN.
* Customers who self-host mParticle via npm - You should add @mparticle/web-braze-kit version 6.0.0 or greater in your package.json.  You must also select `Version 6` under `Braze Web SDK Version`  in the Braze connection settings.
* Customers who load mParticle via snippet/CDN - You must  select `Version 6` under `Braze Web SDK Version`  in the Braze connection settings.

Note that the following is only one example.  Everywhere you manually call `braze` needs to be updated similar to the below. If you are using NPM, you can skip to step 3.  Please be sure to test your site fully in development prior to releasing.

* Step 1: Legacy code sample. Find all the places where your code references `braze` directly.  Braze may have renamed or removed APIs between V5 and V6:
```javascript

> window.braze.someV5Method();

Reviewer: `someV5Method` looks odd. Would need to find either an alternative method or can remove the section all together since looking the changelog they removed support for `displays` and `destroyFeed()` completely. I would suggest removing this section and adding it on line 8 where you mention API changes

```

Step 2: Roll out code changes prior to opting in to V6
```javascript
if (typeof window.braze.someV6Method === 'function') {
	window.braze.someV6Method();
} else {
	window.braze.someV5Method();
}
```
Step 3: Whether you are using the snippet or self hosting, you need to navigate to your Braze connection settings and select `Version 6` from the `Braze Web SDK Version` drop down.

Step 4: After you opt in, you can simplify your code. We recommend testing and waiting at least 24 hours between opting in and removing backward-compatibility shims, and doing thorough testing of your application in a development environment to ensure everything is working:
```javascript

> window.braze.someV6Method();

Reviewer: same comment as above for `someV6Method`

You can opt into the latest major version of the Braze Web SDK whether you implement mParticle's Web SDK using npm or our snippet/CDN.
* Customers who self-host mParticle via npm - You should add @mparticle/web-braze-kit version 6.0.0 or greater in your package.json.  You must also select `Version 6` under `Braze Web SDK Version`  in the Braze connection settings.
* Customers who load mParticle via snippet/CDN - You must  select `Version 6` under `Braze Web SDK Version`  in the Braze connection settings.

Step 1: Whether you are using the snippet or self hosting, you need to navigate to your Braze connection settings and select `Version 6` from the `Braze Web SDK Version` drop down.

Step 2: Remove or replace any calls to removed APIs (e.g. News Feed methods such as `destroyFeed()`, `toggleFeed()`, `showFeed()`) and update any renamed APIs per the migration guide. We recommend testing thoroughly in a development environment before releasing.

Step 3: Push Notifications via service-worker.js

> If you use Push Notifications, we have updated the `service-worker.js` file.  In our testing, Braze's push notifications work as expected regardless of what version of the service-worker is used, but we recommend updating this file to ensure future compatibility.  In your `service-worker.js` file, update the code to reference `https://static.mparticle.com/sdk/js/braze/service-worker-6.5.0.js` instead of `https://static.mparticle.com/sdk/js/braze/service-worker-5.5.0.js`.  Your `service-worker.js` file should now contain:

Reviewer 2: I know this is legacy, but I'd suggest an update without saying "we tested all versions".  
```suggestion
If you use Push Notifications, your `service-worker.js` file should be updated to reference `https://static.mparticle.com/sdk/js/braze/service-worker-6.5.0.js` instead of `https://static.mparticle.com/sdk/js/braze/service-worker-5.5.0.js`.  Your `service-worker.js` file should now contain:
```
